src/transport/HighPressureGasTransport.cpp

```diff
@@ -109,216 +180,141 @@ double HighPressureGasTransport::thermalConductivity()
-{
-    // Method for MultiTransport class:
-    // solveLMatrixEquation();
-    // const double c = 1.6/GasConstant;
-    // for (size_t k = 0; k < m_nsp; k++) {
-    // dt[k] = c * m_mw[k] * m_molefracs[k] * m_a[k];
-    // }
-    throw NotImplementedError("HighPressureGasTransport::getThermalDiffCoeffs");
-}
-
 void HighPressureGasTransport::getBinaryDiffCoeffs(const size_t ld, double* const d)
 {
-    vector<double> PcP(5);
     size_t nsp = m_thermo->nSpecies();
     vector<double> molefracs(nsp);
     m_thermo->getMoleFractions(&molefracs[0]);
 
     update_T();
-    // Evaluate the binary diffusion coefficients from the polynomial fits.
-    // This should perhaps be preceded by a check to see whether any of T, P, or
-    //   C have changed.
-    //if (!m_bindiff_ok) {
-    updateDiff_T();
-    //}
-    if (ld < nsp) {
-        throw CanteraError("HighPressureGasTransport::getBinaryDiffCoeffs",
-                           "ld is too small");
+    // If necessary, evaluate the binary diffusion coefficients from the polynomial fits
+    if (!m_bindiff_ok) {
+        updateDiff_T();
+    }
+    if (ld < m_nsp) {
+        throw CanteraError("HighPressureGasTransport::getBinaryDiffCoeffs", "ld is too small");
     }
+
     double rp = 1.0/m_thermo->pressure();
     for (size_t i = 0; i < nsp; i++) {
         for (size_t j = 0; j < nsp; j++) {
             // Add an offset to avoid a condition where x_i and x_j both equal
-            // zero (this would lead to Pr_ij = Inf):
+            // zero (this would lead to Pr_ij = Inf).
             double x_i = std::max(Tiny, molefracs[i]);
             double x_j = std::max(Tiny, molefracs[j]);
 
-            // Weight mole fractions of i and j so that X_i + X_j = 1.0:
+            // Weight mole fractions of i and j so that X_i + X_j = 1.0.
             x_i = x_i/(x_i + x_j);
             x_j = x_j/(x_i + x_j);
 
-            //Calculate Tr and Pr based on mole-fraction-weighted crit constants:
+            // Calculate Tr and Pr based on mole-fraction-weighted critical constants.
             double Tr_ij = m_temp/(x_i*Tcrit_i(i) + x_j*Tcrit_i(j));
             double Pr_ij = m_thermo->pressure()/(x_i*Pcrit_i(i) + x_j*Pcrit_i(j));
 
+            // Calculate the parameters for Takahashi correlation
             double P_corr_ij;
-            if (Pr_ij < 0.1) {
-                // If pressure is low enough, no correction is needed:
-                P_corr_ij = 1;
-            }else {
-                // Otherwise, calculate the parameters for Takahashi correlation
-                // by interpolating on Pr_ij:
-                P_corr_ij = setPcorr(Pr_ij, Tr_ij);
-
-                // If the reduced temperature is too low, the correction factor
-                // P_corr_ij will be < 0:
-                if (P_corr_ij<0) {
-                    P_corr_ij = Tiny;
-                }
-            }
+            P_corr_ij = takahashi_correction_factor(Pr_ij, Tr_ij);
```

Comment:
You may as well declare the type on the same line
```suggestion
            double P_corr_ij = takahashi_correction_factor(Pr_ij, Tr_ij);
```